web-client/src/modules/create/components/NewAddressModal.tsx

```diff
@@ -0,0 +1,158 @@
+import { Checkbox, Col, Form, Input, Modal, Row } from 'antd';
+import { useForm } from 'antd/lib/form/Form';
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch, useSelector } from 'react-redux';
+import { GeocoderComponent } from 'src/components/Geocoder/Geocoder';
+import { updateUserPrivilegedInformation } from 'src/ducks/profile/actions';
+import { IUserAddress } from 'src/models/users/privilegedInformation';
+import { AppState } from 'src/store';
+
+const ModalForm: React.FC<ModalFormProps> = ({
+  geocode,
+  visible,
+  closeModal,
+  modalSuccess,
+}) => {
+  const [form] = useForm();
+  const dispatch = useDispatch();
+
+  const [shouldSave, setShouldSave] = useState<boolean>(false);
+
+  const addresses = useSelector(
+    (state: AppState) => state.profile.privilegedInformation?.addresses,
+  );
+  const privilegedInformation = useSelector(
+    (state: AppState) => state.profile.privilegedInformation,
+  );
+  const userId = useSelector((state: AppState) => state.auth.user?.uid);
+
+  const handleFinish = () => {
+    geocode({
+      name: form.getFieldValue('name'),
+      address1: form.getFieldValue('address1'),
+      address2: form.getFieldValue('address2'),
+      city: form.getFieldValue('city'),
+      state: form.getFieldValue('state'),
+      postalCode: form.getFieldValue('postalCode'),
+      country: form.getFieldValue('country'),
+    })
+      .then(geocodedResult => {
+        modalSuccess(geocodedResult);
+        if (shouldSave) {
+          if (privilegedInformation && userId) {
+            const addressesToSave = {
+              ...addresses,
+            };
+            addressesToSave[geocodedResult.name] = geocodedResult;
+            privilegedInformation.addresses = addressesToSave;
+            dispatch(
+              updateUserPrivilegedInformation(userId, privilegedInformation),
+            );
+          }
+        }
+      })
+      .catch(error => {
+        // eslint-disable-next-line no-console
+        console.error('could not geocode', error);
```

Comment:
Should this failure be passed back to the container? I think currently if this fails, the modal will just remain visible. passing the address to teh container would allow you to handle this case there.

Reply:
Fixed. onGeocodeFail() defined in CreatePostContainer